### PR TITLE
[Windows] Fix root folder filter in native file dialog.

### LINKED ISSUE #98832: Native FileDialog does not open on Windows when using FILE_MODE_OPEN_DIR in some 4.4 dev builds
### Tested versions

- Reproducible in: v4.4.dev2.official [97ef3c837] and v4.4.dev3.official [f4af8201b]
- Not reproducible in: v4.4.dev1.official [28a72fa43] and v4.3.stable.official [77dcf97d8]

### System information

Godot v4.4.dev3 - Windows 10.0.22631 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 Ti (NVIDIA; 32.0.15.6094) - AMD Ryzen 7 5700G with Radeon Graphics (16 threads)

### Issue description

In current development builds of Godot 4.4 _(dev2 onwards)_, opening a native FileDialog on Windows fails if the file mode is `FileDialog.FILE_MODE_OPEN_DIR`. Specifically, the game window loses focus for a moment before immediately re-gaining focus, almost as if the separate process immediately exits.

This behavior does not exist and functions correctly when:
- Using other file modes
- Used in Godot 4.4.dev1 or Godot 4.3.
- Calling `DisplayServer.file_dialog_show()` with `DisplayServer.FILE_DIALOG_MODE_OPEN_DIR` in any version

I do not have the means to test behavior on macOS or Linux, so I'm unsure if those operating systems are affected.

### Steps to reproduce

To observe the behavior, the following code can be used:
```gd
var dialog = FileDialog.new()
dialog.use_native_dialog = true
dialog.file_mode = FileDialog.FILE_MODE_OPEN_DIR
# ... bring FileDialog to scene tree ...
dialog.popup_centered()
```

The DisplayServer can be used as a baseline and works in all versions:
```gd
DisplayServer.file_dialog_show(
    "Test Dialog", "", "", false, 
    DisplayServer.FILE_DIALOG_MODE_OPEN_DIR,
    PackedStringArray(), func c(_a, _b, _c): pass
)
```

### Minimal reproduction project (MRP)

[OpenDirTest.zip](https://github.com/user-attachments/files/17625616/OpenDirTest.zip)

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/98832

### CODE REVIEW

File: platform/windows/display_server_windows.cpp [hunk 1]
```diff
@@ -571,10 +571,7 @@ void DisplayServerWindows::_thread_fd_monitor(void *p_ud) {
 			}
 		}
 		dir = dir.simplify_path();
-		dir = dir.replace("/", "\\");
-		if (!dir.is_network_share_path() && !dir.begins_with(R"(\\?\)")) {
-			dir = R"(\\?\)" + dir;
-		}
+		dir = dir.trim_prefix(R"(\\?\)").replace("/", "\\");
```
Comment (riverar) [inline platform/windows/display_server_windows.cpp:574]: This will likely break extended path support (e.g., long paths), something to be aware of.

Comment (bruvzg) [reply platform/windows/display_server_windows.cpp:574]: `SHCreateItemFromParsingName` do not have any long path support, so it won't break anything.

Comment (riverar) [reply platform/windows/display_server_windows.cpp:574]: Ah, missed that function below. Sounds good.

**Comment by bruvzg** [comment]
Superseded by https://github.com/godotengine/godot/pull/99652

**Pull request closed without merging**